### PR TITLE
story/EUI-7113 roles-and-access

### PR DESCRIPTION
**Pull request by Kartheek-S**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-7113

### Change description ###
Work allocation R3.5 - Feature toggle R3.5

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: api/workAllocation/index.ts [hunk 1]
```diff
@@ -140,28 +141,91 @@ export async function searchTask(req: EnhancedRequest, res: Response, next: Next
...
     res.status(status);
     // Assign actions to the tasks on the data from the API.
     let returnData;
     if (data) {
       // Note: TaskPermission placed in here is an example of what we could be getting (i.e. Manage permission)
       // These should be mocked as if we were getting them from the user themselves
-      returnData = { tasks: assignActionsToTasks(data.tasks, req.body.view, currentUser), total_records: data.total_records };
+      if (refined) {
+        data = mockTaskPermissions(data);
+      }
```
Comment (connorpgpmcelroy) [inline api/workAllocation/index.ts:167]: Need to remove mockTaskPermissions call now btw please

File: src/app/app.constants.ts [hunk 2]
```diff
@@ -4,12 +4,14 @@ import { ApplicationTheme, NavigationItem } from './models/theming.model';
...
   waServiceConfig: 'wa-service-config',
+  updatedTaskPermissionsFeature: 'updated-task-permission',
   noticeOfChange: 'MC_Notice_of_Change',
   waMvpPaginationFeature: 'mc-mvp-wa-pagination',
   userTypeRoles: 'mc-user-type-roles',
   booking: 'mc-booking-active',
   mcHearingsFeature: 'mc-hearings-jurisdictions',
-  excludedRolesForCaseTabs: 'mc-excluded-roles-case-tabs'
+  excludedRolesForCaseTabs: 'mc-excluded-roles-case-tabs',
+  updatedTaskPermission: 'updated-task-permission',
```
Comment (connorpgpmcelroy) [inline src/app/app.constants.ts:14]: two updatedTaskPermission feature calls here - need to be merged

File: src/cases/containers/roles-and-access-container/roles-and-access-container.component.ts [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-import {UserDetails} from '../../../app/models/user-details.model';
-import {SessionStorageService} from '../../../app/services';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { CaseView } from '@hmcts/ccd-case-ui-toolkit';
+import { FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
+import { Store } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import { first, map, mergeMap, tap } from 'rxjs/operators';
+import { AppConstants } from 'src/app/app.constants';
```
Comment (connorpgpmcelroy) [inline src/cases/containers/roles-and-access-container/roles-and-access-container.component.ts:8]: src needs to be replaces

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Changes required but looks in the main good to me, thanks for replicating the task permission changes

**Comment by Kartheek-S** [comment]
Changes merged into [EUI-7032](https://github.com/hmcts/rpx-xui-webapp/pull/2516)

**Pull request closed without merging**